### PR TITLE
Added updating of objects by CSV

### PR DESCRIPTION
**Pull request by doronlevari**

### CODE REVIEW

File: object_update_csv/cdo.update.object [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+[ ${1} ] || fail "${0} <input CSV file name> [doit]"
+[ -f ${1} ] || fail "input file not found: ${1}"
+
+# validate OAUTH token
+API_URL="https://www.defenseorchestrator.com/aegis/rest/v1/services"
+curl -s -o /dev/null -f -H "Content-Type: application/json" -H "Authorization: bearer ${OAUTH}" -X GET "${API_URL}" || failexit "\"${OAUTH}\" is an invalid OAuth token"
+
+[ ${2} ] || log "****** DRY RUN **********"
+
+#while IFS=, read -r DEVICE_NAME OBJECT_NAME NEW_SOURCE NEW_DEST alltherest
```
Comment (jperry) [inline object_update_csv/cdo.update.object:23]: delete?

File: object_update_csv/cdo.update.object [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  OBJECT_NAME="${array[1]}"
+  NEW_VALUE="${array[2]}"
+  NEW_DEST="${array[3]}"
+
+  log "processing line: ${DEVICE_NAME} ${OBJECT_NAME} ${NEW_VALUE}"
+
+  # handle range values. e.g. 10.10.238.16-10.10.238.17
+  NEW_SOURCE=`echo "${NEW_VALUE}" | cut -d '-' -f 1`
+  NEW_DEST=`echo "${NEW_VALUE}" | cut -d '-' -f 2`
+  # if no range, then both values are same, have NEW_DEST be null, otherwise quote it, I won't add quotes in the concat
```
Comment (jperry) [inline object_update_csv/cdo.update.object:39]: delete?

File: object_update_csv/cdo.update.object [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    log "   could not find device by name ${DEVICE_NAME}"
+    continue
+  fi
+
+  log "   (with device UID): ${DEVICE_NAME} ${DEVICE_UID} ${OBJECT_NAME} ${NEW_SOURCE} ${NEW_DEST}"
+
+  # find and retrieve the object
+  OBJECT_URL="${API_URL}/targets/objects"
+  OBJECT_SEARCH_URL="${OBJECT_URL}?q=name:${OBJECT_NAME}%20AND%20deviceUid:${DEVICE_UID}"
+  # curl -s -f -H "Content-Type: application/json" -H "Authorization: bearer ${OAUTH}" -X GET "${OBJECT_SEARCH_URL}" | jq
```
Comment (jperry) [inline object_update_csv/cdo.update.object:61]: delete?

File: object_update_csv/cdo.update.object [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+  if [ "x${OBJECT_UID}x" == "xx" ] || [ "x${OBJECT_UID}" == "xnull" ]
+  then
+    log "   could not find object by name ${OBJECT_NAME}"
+    continue
+  fi
+
+  log "   (with object UID): ${DEVICE_NAME} ${DEVICE_UID} ${OBJECT_NAME} ${OBJECT_UID} ${NEW_SOURCE} ${NEW_DEST}"
+
+  # echo "[ { \"@type\": \"NetworkContent\", \"sourceElement\": \"${NEW_VALUE}\", \"destinationElement\": null, \"wildcardMaskElement\": null } ]" > body.json
```
Comment (jperry) [inline object_update_csv/cdo.update.object:73]: delete?

File: object_update_csv/cdo.update.object [hunk 5]
```diff
@@ -0,0 +1,86 @@
+#!/bin/sh 
+
+#Usage:
+#export OAUTH=my_API_token
+#bash cdo.update.object input.csv doit
+ 
+# log  () { echo $(date +"[%F %R:%S] ") "$@"; }
```
Comment (jperry) [inline object_update_csv/cdo.update.object:7]: delete?

File: object_update_csv/cdo.update.object [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+failexit () { log "failure: $@"; exit 1; }
+fail () { log "failure: $@"; }
+ 
+which -s jq || fail "jq must be installed"
+[ ${OAUTH} ] || fail "You must specify OAuth token in the \$OAUTH system variable"
+[ ${1} ] || fail "${0} <input CSV file name> [doit]"
+[ -f ${1} ] || fail "input file not found: ${1}"
+
+# validate OAUTH token
+API_URL="https://www.defenseorchestrator.com/aegis/rest/v1/services"
```
Comment (jperry) [inline object_update_csv/cdo.update.object:18]: should you make the endpoint configurable?  Meaning for EU or APJ?

File: object_update_csv/cdo.update.object [hunk 7]
```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
```
Comment (jperry) [inline object_update_csv/cdo.update.object:1]: Also rename this file to end in .sh, standard for shell scripts.

File: object_update_csv/cdo.update.object (hunk 1 above)
Comment (doronlevari) [reply object_update_csv/cdo.update.object]: done

File: object_update_csv/cdo.update.object (hunk 3 above)
Comment (doronlevari) [reply object_update_csv/cdo.update.object]: done

File: object_update_csv/cdo.update.object (hunk 4 above)
Comment (doronlevari) [reply object_update_csv/cdo.update.object]: done

File: object_update_csv/cdo.update.object (hunk 5 above)
Comment (doronlevari) [reply object_update_csv/cdo.update.object]: done

File: object_update_csv/cdo.update.object (hunk 6 above)
Comment (doronlevari) [reply object_update_csv/cdo.update.object:17]: one time we'll do that to all scripts here..... I don't have the time to do it now

File: object_update_csv/cdo.update.object (hunk 7 above)
Comment (doronlevari) [reply object_update_csv/cdo.update.object:1]: ok.

**Pull request merged**